modules/standard/IO.chpl

```diff
@@ -2889,20 +2894,20 @@ This function is equivalent to calling :proc:`open` with ``iomode.cwr`` and then
 */
 proc openwriter(path:string,
                 param kind=iokind.dynamic, param locking=true,
-                start:int(64) = 0, end:int(64) = max(int(64)),
                 hints = ioHintSet.empty)
     : channel(true, kind, locking) throws {
-  return openwriterHelper(path, kind, locking, start, end, hints);
+  return openwriterHelper(path, kind, locking, hints=hints);
 }
 
 pragma "last resort"
 deprecated "The 'iohints' type is deprecated; please use a variant of 'openwriter' that takes an 'ioHintSet' instead."
 proc openwriter(path:string,
                 param kind=iokind.dynamic, param locking=true,
                 start:int(64) = 0, end:int(64) = max(int(64)),
-                hints:iohints=IOHINT_NONE)
+                hints:iohints)
```

Comment:
Not sure I'm understanding the logic for removing the default value on the hints argument. Is it because `openwriterHelper` has a default value for that argument anyway?

Reply:
It's because the default value makes it conflict with the newly added deprecated version of openwriter.  Without requiring the hints argument for this version, the compiler doesn't have a way to distinguish between the two candidates when no value is provided for that argument